http/util/src/main/java/org/wildfly/security/http/util/SetRequestInformationMechanismFactory.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.http.util;
+
+import org.wildfly.security.auth.server.RequestInformationCallback;
+import org.wildfly.security.http.HttpAuthenticationException;
+import org.wildfly.security.http.HttpServerAuthenticationMechanism;
+import org.wildfly.security.http.HttpServerAuthenticationMechanismFactory;
+import org.wildfly.security.http.HttpServerRequest;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import java.util.Map;
+
+import static org.wildfly.security.http.util.ElytronMessages.log;
+
+/**
+ * A wrapper {@link HttpServerAuthenticationMechanismFactory} that sets the request information using the current authentication request.
+ *
+ * @author <a href="mailto:dvilkola@redhat.com">Diana Krepinska</a>
+ */
+public class SetRequestInformationMechanismFactory implements HttpServerAuthenticationMechanismFactory {
```

Comment:
I think this should be called `RequestInformationCallbackServerMechanismFactory` to be consistent with other the factories, right?

Reply:
I see now, we also have a `SetMechanismInformationMechanismFactory` so that's probably why you named this `SetRequestInformationMechanismFactory`. That's fine then.

Reply:
@fjuma  I renamed this because I was thinking the same before. The `SocketAddressCallbackServerMechanismFactory` also uses the Callback in its name. I think the `RequestInformationCallbackServerMechanismFactory` is more clear